lib/registry/registry.rb

```diff
@@ -257,7 +257,7 @@ def doreq(type,url,stream=nil,payload=nil)
         raise DockerRegistry2::NotFound, error
       rescue RestClient::Unauthorized => e
         header = e.response.headers[:www_authenticate]
-        method = header.downcase.split(' ')[0]
+        method = header.to_s.downcase.split(' ')[0]
```

Comment:
Does this work, then? if `header` is undefined, will `header.to_s` force it into a string of `""` and then avoid the error?

Reply:
It does work as the only use of `method` in the method is to check against the case statement, and the desired behavior is that it should match neither authentication type, and then fall into the `else` and raise a `RegistryUnknownException`